Scripts/Battle/battle_manager.gd

```diff
@@ -127,7 +136,25 @@ func start_battle(new_armies : Array[Army], battle_map : DataBattleMap, \
 	selected_unit = null
 	battle_ui.load_armies(_battle_grid.armies_in_battle_state)
 
-	on_turn_started(_battle_grid.get_current_player())
+	if not _batch_mode:
+		on_turn_started(_battle_grid.get_current_player())
+
+
+func force_battle_state(armies : Array[Army], \
+		map_data : DataBattleMap, \
+		battle_state : SerializableBattleState, x_offset : float):
```

Comment:
i would not add x_offset as a parameter
x_offset is hacky way to make battles not collide with world, its not a real parameter

also shouldn't SerializableBattleState contain map and armies...
seams weerd to have separate parameters

Reply:
ok this is not a force_battle_state
it is alternative start_battle with state to deserialize...

Reply:
The parameter `armies` is separate because battle state is optional and has these values in other format. `map_data` is in same format, good point. I didn't have better idea yet for it, so I left it this way. Probably I will fully merge start_battle and force_battle_state, but I'm not sure yet. Also, these parameters like `world_armies` etc. can be rearranged.